binary_search_tree/binary_search_tree.py

```diff
@@ -5,13 +5,38 @@ def __init__(self, value):
     self.right = None
 
   def insert(self, value):
-    pass
+    if value == self.value:
+      return "Value found"
+    if value < self.value:
+      if self.left is None:
+        self.left = BinarySearchTree(value)
+      else:
+        self.left.insert(value)
+    elif value > self.value:
+      if self.right is None:
+        self.right = BinarySearchTree(value)
+      else:
+        self.right.insert(value)
+    else:
+      self.value = value
 
   def contains(self, target):
-    pass
+    if self.value == target:
+      return True
+    else:
+      if target < self.value:
+        self.left = BinarySearchTree(self.left)
+        self.left.contains(target)
+      else:
+        self.right = BinarySearchTree(self.right)
```

Comment:
I'm curious if you actually have to declare this each time? Because I think self is already a node and so self.right should already have access to the contains method.